### PR TITLE
doc/dev/msgr2: add state machine diagrams

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
not sure if it's helpful or not.

File: doc/dev/msgr2.rst [hunk 1]
```diff
@@ -838,3 +838,56 @@ _____________________________________
                 |                   |
 
 
+.. graphviz::
```
Comment (batrick) [inline doc/dev/msgr2.rst:841]: !! That's so much easier than what I have here

https://github.com/ceph/ceph/blob/ea706273b53ea25232982fc485cbbf056f7203b6/doc/cephfs/mds-state-diagram.dot

Including a Makefile and svg in git. I'll have to make a note to migrate that into the rst.

Comment (batrick) [reply doc/dev/msgr2.rst:841]: https://tracker.ceph.com/issues/50865

Comment (tchaikov) [reply doc/dev/msgr2.rst:841]: @batrick sphinx also allows us to include a dot file by referencing it. see https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html. i think the upside of using a dot file is that it's easier to be process with external tools.

**Comment by rzarzynski** [comment]
@tchaikov: is there a straightforward way to visualize the diagram before merge?

**Comment by tchaikov** [comment]
@rzarzynski https://ceph--41382.org.readthedocs.build/en/41382/dev/msgr2/#example-of-protocol-interaction-wip

**Comment by tchaikov** [comment]
i wanted to use tooltip to explain where update the peer address and connection's feature bits are updated, but seems it does not work.

**Review by jdurgin — APPROVED** [review]
awesome, this is way easier to understand visually!

**Pull request merged**